### PR TITLE
[Restructure/plasma] Refactor the simulation state initialization

### PR DESCRIPTION
**Pull request by wkerzendorf**
Is based on the restructure radiation field PR. #2613

### CODE REVIEW

**Comment by andrewfullard** [comment]
The general idea is good but I feel like we are back in `parse_input` territory, same as with the I/O before my refactoring. The top-level modules (atom data, plasma, transport etc) should be initialized in separate files with all their associated functions.

**Comment by andrewfullard** [comment]
My previous comment still stands I think. This is also somewhat difficult to review given it is based on another PR.

**Comment by wkerzendorf** [comment]
@andrewfullard - I presume you want me to move some of these functions to `io` - https://github.com/tardis-sn/tardis/pull/2664/files#diff-24577c067ab859927ea5939db7af1e47be15e9b6d48c73fc8a9177ce332b0dd1 ?

**Comment by andrewfullard** [comment]
> @andrewfullard - I presume you want me to move some of these functions to `io` - https://github.com/tardis-sn/tardis/pull/2664/files#diff-24577c067ab859927ea5939db7af1e47be15e9b6d48c73fc8a9177ce332b0dd1 ?

Yes please. That is what they are doing, after all- reading stuff in.

File: tardis/simulation/initialization.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+from tardis.io.atom_data.base import AtomData
+from tardis.io.model.parse_packet_source_configuration import (
+    initialize_packet_source,
+)
+from tardis.model import SimulationState
+
+logger = logging.getLogger(__name__)
+
+
+def initialize_atom_data(config, atom_data=None):
```
Comment (andrewfullard) [inline tardis/simulation/initialization.py:13]: `parse_atom_data` in `io/model` (dir probably needs a rename)

File: tardis/simulation/initialization.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                "please see https://github.com/tardis-sn/tardis-refdata/tree/master/atom_data"
+                " for the most recent version.",
+            )
+            raise
+
+    return atom_data
+
+
+def initialize_simulation_state(
+    config, packet_source, legacy_mode_enabled, kwargs, atom_data
```
Comment (andrewfullard) [inline tardis/simulation/initialization.py:61]: `parse_simulation_state` in `io/model`

**Comment by wkerzendorf** [comment]
@andrewfullard - moved

**Comment by andrewfullard** [comment]
Looks good- is this ready to review?

**Comment by wkerzendorf** [comment]
> Looks good- is this ready to review?

I want to say yes

**Review by andrewfullard — APPROVED** [review]
Looks good! Pending codestyle. I guess this makes Simulation approach the workflow in my branch.

File: tardis/io/model/parse_packet_source_configuration.py [hunk 3]
```diff
@@ -46,7 +46,7 @@ def initialize_packet_source(packet_source, config, geometry):
     return packet_source
 
 
-def parse_packet_source_from_config(config, geometry, legacy_mode_enabled):
+def parse_packet_source_from_config(config, geometry, enable_legacy_mode):
```
Comment (Rodot-) [inline tardis/io/model/parse_packet_source_configuration.py:49]: Why make this change?  Don't we usually want booleans to be in a format that reads like `flag is enabled` or `if flag_enabled`?

**Review by Rodot- — DISMISSED** [review]
Looks good to me

**Pull request merged**